### ISSUE TITLE
Break out underlying logic from _check_condition and _check_principal into separate functions for library consumption

### ISSUE THREAD

**Issue by kmcquade** [duo-labs/parliament#89]
The underlying logic behind `_check_principal` and `_check_condition` is gold. However, I would like to use Parliament to validate the logic of IAM policy subsections - specifically the principals block and condition blocks - but as-is, Parliament does not expose that underlying logic. I'd like to run this idea by you before making any kind of PR, so that if I do, I take an approach that is mutually agreeable and fits with your vision for Parliament.

For instance - this code snippet is part of the `_check_condition` method under the Statement class. I'd like to turn this into a method titled `is_valid_operator` instead.

Existing code:

```python
        operator_type_requirement = None
        for documented_operator in OPERATORS:
            op = documented_operator.lower()
            if operator.lower() in [
                op,
                op + "ifexists",
                "forallvalues:" + op,
                "foranyvalue:" + op,
                "forallvalues:" + op + "ifexists",
                "foranyvalue:" + op + "ifexists",
            ]:
                operator_type_requirement = OPERATORS[documented_operator]
                break
```

Implementation-wise, we'd have to decouple the logic from the `self.add_finding` calls for this to be successful. But it would be extremely helpful to other libraries/tools that want to use Parliament to parse/validate subsections of IAM policies.

Let me know what you think.

**Comment by 0xdabbad00** [comment]
I'd be fine with `_check_principal` and `_check_condition` being renamed to remove the leading `_` so they are public, and moving the code in `_check_condition` for the operators to their own function.

**Comment by kmcquade** [comment]
I'd want to do it for the rest of the checks under the `_check_principal` and `_check_condition` methods too. For example - 

`check_bad_pattern_for_mfa`
`is_mismatched_type_set_to_null`
`check_mismatched_global_condition_type`
`check_mismatched_service_condition_type`
`check_action_condition_not_found`
`check_unknown_condition_for_action`

And then based on the returned response for each, the main _check_condition function would do `self.add_finding`. If someone consuming this library, such as myself, wanted to leverage the logic, they could just call those methods and then turn them into raised Exceptions instead.

Does that make sense?

**Comment by 0xdabbad00** [comment]
Makes sense. As much as possible I would try to have your external library just call the highest level function it can so it doesn't have to rely too much on these child functions.  Also, the `check_bad_pattern_for_mfa` I probably should move somewhere else, like maybe the community auditors even.
